sit-common/src/main/java/com/sit/common/api/interceptor/NetworkConnectionInterceptor.kt

```diff
@@ -0,0 +1,45 @@
+package com.sit.common.api.interceptor
+
+import android.content.Context
+import android.net.ConnectivityManager
+import android.net.NetworkCapabilities
+import com.sit.common.utils.Constant
+import com.sit.common.utils.NoInternetException
+import okhttp3.Interceptor
+import okhttp3.Response
+import java.net.InetAddress
+import java.net.UnknownHostException
+
+class NetworkConnectionInterceptor(
+    context: Context
+) : Interceptor {
+
+    private val applicationContext = context.applicationContext
+
+    override fun intercept(chain: Interceptor.Chain): Response {
+        if (!isInternetAvailable())
+            throw NoInternetException(Constant.NO_INTERNET)
```

Comment:
Please use strings from the strings.xml file so we can adjust the messages according to the app's needs.